### PR TITLE
New chapter: advanced scenarios

### PR DESCRIPTION
**Pull request by derisen**
## Purpose

* New sample: `7-AdvancedScenarios/1-call-api-obo` - Angular SPA calling Microsoft Graph via .NET Core web API using OBO
* New sample: `7-AdvancedScenarios/2-call-api-pop` - Angular SPA calling .NET Core web API using Proof of Possession tokens

## Does this introduce a breaking change?

```
[ ] Yes
[ x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe: **new sample**
```

## How to Test
*  Get the code

```
git clone https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial.git
git checkout dev
cd ms-identity-javascript-angular-tutorial/7-AdvancedScenarios/<sample_folder>
cd SPA
npm install
cd ../TodoListAPI
dotnet restore
```

## What to Check
Verify that the following are valid
* Run in Chrome, Firefox or Edge
* Run in incognito mode

cc @j-mantu

### CODE REVIEW

File: 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      "RepositoryUrl": "ms-identity-javascript-angular-tutorial",
+      "Endpoint": "AAD v2.0"
+    },
+    "AADApps": [
+      {
+        "Id": "service",
+        "Name": "msal-dotnet-api",
+        "Kind": "WebApi",
+        "Audience": "AzureADMyOrg",
+        "HomePage": "https://localhost::44372/api/todolist"
```
Comment (Shama-K) [inline 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/sample.json:16]: HomePage value needs update: "https://localhost:44372/api/todolist"

File: 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/Configure.ps1 [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    $tenantName =  ($tenant.VerifiedDomains | Where { $_._Default -eq $True }).Name
+
+    # Get the user running the script to add the user as the app owner
+    $user = Get-AzureADUser -ObjectId $creds.Account.Id
+
+   # Create the service AAD application
+   Write-Host "Creating the AAD application (msal-dotnet-api)"
+   # create the application 
+   $serviceAadApplication = New-AzureADApplication -DisplayName "msal-dotnet-api" `
+                                                   -HomePage "https://localhost::44372/api/todolist" `
```
Comment (Shama-K) [inline 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/Configure.ps1:203]: HomePage value needs update: "https://localhost:44372/api/todolist"

File: 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/sample.json (hunk 1 above)
Comment (derisen) [reply 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/sample.json:16]: done!

File: 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/Configure.ps1 (hunk 2 above)
Comment (derisen) [reply 7-AdvancedScenarios/2-call-api-pop/AppCreationScripts/Configure.ps1:203]: done!

File: 7-AdvancedScenarios/2-call-api-pop/README.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+or download and extract the repository .zip file.
+
+> :warning: To avoid path length limitations on Windows, we recommend cloning into a directory near the root of your drive.
+
+### Step 2. Install .NET Core API dependencies
+
+```console
+    cd ms-identity-javascript-angular-tutorial
+    cd 3-Authorization-II/1-call-api/API/TodoListAPI
```
Comment (Shama-K) [inline 7-AdvancedScenarios/2-call-api-pop/README.md:61]: Path should be: cd 7-AdvancedScenarios/2-call-api-pop/API/TodoListAPI

Comment (derisen) [reply 7-AdvancedScenarios/2-call-api-pop/README.md:61]: oops, done!

File: 7-AdvancedScenarios/2-call-api-pop/README.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+```console
+    cd SPA
+    npm start
+```
+
+In a separate console window, execute the following commands:
+
+```console
+    cd API
```
Comment (Shama-K) [inline 7-AdvancedScenarios/2-call-api-pop/README.md:204]: Required path: cd API/TodoListAPI

Comment (derisen) [reply 7-AdvancedScenarios/2-call-api-pop/README.md:204]: fixed!

**Comment by idg-sam** [comment]
`Could not load file or assembly 'System.Net.Sockets, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. A device attached to the system is not functioning.`
unable to do `dotnet restore` in API/TodoListAPI folder.

File: 7-AdvancedScenarios/1-call-api-obo/README.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+or download and extract the repository .zip file.
+
+> :warning: To avoid path length limitations on Windows, we recommend cloning into a directory near the root of your drive.
+
+### Step 2. Install .NET Core API dependencies
+
+```console
+    cd ms-identity-javascript-angular-tutorial
+    cd 3-Authorization-II/1-call-api/API/TodoListAPI
```
Comment (Shama-K) [inline 7-AdvancedScenarios/1-call-api-obo/README.md:60]: Path: 7-AdvancedScenarios/1-call-api-obo/API

File: 7-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1 [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+
+
+   # Update config file for 'service'
+   $configFile = $pwd.Path + "\..\API\appsettings.json"
+   Write-Host "Updating the sample code ($configFile)"
+   $dictionary = @{ "Domain" = $tenantName;"ClientId" = $serviceAadApplication.AppId;"ClientSecret" = $serviceAppKey;"TenantId" = $tenantId };
+   UpdateTextFile -configFilePath $configFile -dictionary $dictionary
+
+   # Update config file for 'client'
+   $configFile = $pwd.Path + "\..\SPA\src\app\authConfig.ts"
```
Comment (Shama-K) [inline 7-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1:405]: File name is: auth-config.ts

File: 7-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+          "key": "TenantId",
+          "value": "$tenantId"
+        }
+      ]
+    },
+
+    {
+      "App": "client",
+      "SettingKind": "Replace",
+      "SettingFile": "\\..\\SPA\\src\\app\\authConfig.ts",
```
Comment (Shama-K) [inline 7-AdvancedScenarios/1-call-api-obo/AppCreationScripts/sample.json:77]: File name: auth-config.ts

File: 7-AdvancedScenarios/1-call-api-obo/README.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+       - In the **Delegated permissions** section, select the **Access 'ProfileAPI'** in the list. Use the search box if necessary.
+       - Select the **Add permissions** button at the bottom.
+
+#### Configure the client app (ProfileSPA) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `SPA\src\app\authConfig.ts` file.
```
Comment (Shama-K) [inline 7-AdvancedScenarios/1-call-api-obo/README.md:204]: File name: auth-config.ts

File: 7-AdvancedScenarios/2-call-api-pop/README.md [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+#### Configure the client app (msal-angular-spa) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `SPA\src\app\auth-config.ts` file.
+1. Find the key `Enter_the_Application_Id_Here` and replace the existing value with the application ID (clientId) of `msal-angular-spa` app copied from the Azure portal.
+1. Find the key `Enter_the_Tenant_Info_Here` and replace the existing value with your Azure AD tenant ID.
+1. Find the key `Enter_the_Web_Api_Scope_here` and replace the existing value with Scope.
```
Comment (idg-sam) [inline 7-AdvancedScenarios/2-call-api-pop/README.md:191]:
```suggestion
1. Find the key `Enter_the_Web_Api_Scope_here` and replace the existing value with the scope value you saved earlier (e.g., `api://{your msal-dotnet-api's client-id}/access_as_user`.
```

File: 7-AdvancedScenarios/2-call-api-pop/README.md [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+Were we successful in addressing your learning objective? Consider taking a moment to [share your experience with us](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR73pcsbpbxNJuZCMKN0lURpUOU5PNlM4MzRRV0lETkk2ODBPT0NBTEY5MCQlQCN0PWcu).
+
+## About the code
+
+### Acquiring PoP tokens
+
+### Calling web API using PoP scheme
+
+### Validating PoP tokens
+
```
Comment (idg-sam) [inline 7-AdvancedScenarios/2-call-api-pop/README.md:230]: Resolve this comment once adding about-the-code section in 7-2 POP sample.

**Comment by idg-sam** [comment]
7-2 POP looks good; left a couple nits to address.

File: 7-AdvancedScenarios/1-call-api-obo/README.md [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+
+## We'd love your feedback!
+
+Were we successful in addressing your learning objective? Consider taking a moment to [share your experience with us](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR73pcsbpbxNJuZCMKN0lURpUOU5PNlM4MzRRV0lETkk2ODBPT0NBTEY5MCQlQCN0PWcu).
+
+## About the code
+
+> - Describe where the code uses auth libraries, or calls the graph
+> - Describe specific aspects (e.g. caching, validation etc.)
+
```
Comment (idg-sam) [inline 7-AdvancedScenarios/1-call-api-obo/README.md:265]: resolve this comment once you add about-the-code section to 7-1

**Comment by idg-sam** [comment]
7-1 OBO looks good! I've left just one small reminder to address.

File: 7-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1 (hunk 6 above)
Comment (derisen) [reply 7-AdvancedScenarios/1-call-api-obo/AppCreationScripts/Configure.ps1:405]: oops

**Pull request merged**